activestorage/lib/active_storage/analyzer/image_analyzer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  # Extracts width and height in pixels from an image blob.
+  #
+  # Example:
+  #
+  #   ActiveStorage::Analyzer::ImageAnalyzer.new(blob).metadata
+  #   # => { width: 4104, height: 2736 }
+  #
+  # This analyzer relies on the third-party [MiniMagick]{https://github.com/minimagick/minimagick} gem. MiniMagick requires
+  # the [ImageMagick]{http://www.imagemagick.org} system library. These libraries are not provided by Rails; you must
+  # install them yourself to use this analyzer.
+  class Analyzer::ImageAnalyzer < Analyzer
+    def self.accept?(blob)
+      blob.image?
+    end
+
+    def metadata
+      read_image do |image|
+        { width: image.width, height: image.height }
+      end
+    end
+
+    private
+      def read_image
+        download_blob_to_tempfile do |file|
+          require "mini_magick"
```

Comment:
is `mini_magick` a dependency of Rails now?

Reply:
It’s an optional Active Storage dependency.

Reply:
Can we gracefully fail for this require here telling people to include the gem if they don't have?